plugins/node/opentelemetry-instrumentation-mongodb/src/mongodb.ts

```diff
@@ -197,14 +197,7 @@ export class MongoDBInstrumentation extends InstrumentationBase<
           typeof resultHandler !== 'function' ||
           typeof cmd !== 'object'
         ) {
-          return original.call(
-            this,
-            server,
-            ns,
-            cmd,
-            typeof options === 'function' ? callback : options,
-            callback
-          );
+          return original.call(this, server, ns, cmd, options, callback);
```

Comment:
```suggestion
          return original.apply(this, arguments);
```
Consider callback is not given by caller. We would see `arguments.length === 4` here but mongodb would see `arguments.length === 5`

Reply:
👍 to this

The types might complain so you might even want to use `Reflect.apply`

Reply:
You are right, i fixed it

Reply:
I can't use `arguments` because typescript think its type is `IArguments` instead of the concrete type i wrote. I fixed the `arguments.length` issue thought

Reply:
usually it works also to use rest arguments instead relying on `arguments` because rest arguments can be typed as needed.

Reply:
I'm not sure to see how we can type rest arguments in this case, is there an example somewhere ? thanks

Reply:
According to ts doc it should work, see https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#rest-parameters-with-tuple-types
But not sure if it would really improve the code here as we maybe just move complexity from A to B.

Reply:
I see indeed it would be possible but i also don't think its improve the readability of the code. I think the current version is simple enough (even though it can break if they changes the number of arguments, which should not happen since they have been rewriting from scratch the module recently)